src/executor/current_thread/scheduler.rs

```diff
@@ -154,27 +162,59 @@ where W: Wakeup,
             nodes: List::new(),
         }
     }
-}
 
-impl<T, W: Wakeup> Scheduler<T, W> {
-    /// Advance the scheduler state.
+    pub fn notify(&self) -> NotifyHandle {
+        self.inner.clone().into()
+    }
+
+    pub fn schedule(&mut self, item: Box<Future<Item = (), Error = ()>>) {
```

Comment:
Any way to avoid one of the two allocations here? One from `Arc` and one from the `Box`'ed future.

Reply:
There *might* be a way, but it is actually pretty tricky due to the need for trait objects. I opted to not worry about it for now until it is proven to be an issue.